### PR TITLE
test: always connect peers more realistically

### PR DESCRIPTION
**Pull request by EvanHahn**
_I recommend [reviewing this with whitespace changes disabled](https://github.com/digidem/comapeo-core/pull/856/files?w=1)._

Our tests currently connect `MapeoManager`s in two ways:

1. By starting peer discovery servers and connecting. This is similar to what the real app does.
2. By manually creating streams and connecting them in tests. This is less realistic.

This commit removes the second way because:

- it is less realistic
- it lets us remove some test-only code in the `src/` directory
- it will make an upcoming change easier
- it *used* to be significantly faster, but that's no longer true

I also tried to fix a possible (test-only) race condition, which *could* have been a reason for the less realistic option:

1. Start connecting peers by calling `connectPeers()`. This begins the process of starting peer discovery servers.
2. Disconnect them by calling the callback returned by `connectPeers()`.
3. The peer discovery servers start, and begin connecting. *This is bad* because we already wanted to disconnect!

### CODE REVIEW

File: test-e2e/utils.js [hunk 1]
```diff
@@ -27,50 +27,28 @@ const clientMigrationsFolder = new URL('../drizzle/client', import.meta.url)
 
 /**
  * @param {readonly MapeoManager[]} managers
- * @returns {() => void}
```
Comment (EvanHahn) [inline test-e2e/utils.js:30]: This return type was was just incorrect before, I believe.

File: test-e2e/utils.js [hunk 2]
```diff
@@ -27,50 +27,28 @@ const clientMigrationsFolder = new URL('../drizzle/client', import.meta.url)
...
-              stream.destroy()
-            })
-          )
-        )
+export function connectPeers(managers) {
+  let requestedDisconnect = false
+
+  for (const manager of managers) {
+    manager.startLocalPeerDiscoveryServer().then(({ name, port }) => {
+      if (requestedDisconnect) return
```
Comment (EvanHahn) [inline test-e2e/utils.js:37]: This check (hopefully) fixes a small possible race condition.

File: src/mapeo-project.js [hunk 3]
```diff
@@ -575,29 +574,6 @@ export class MapeoProject extends TypedEmitter {
...
-        const protomux =
-          /** @type {import('protomux')<import('@hyperswarm/secret-stream')>} */ (
-            replicationStream.noiseStream.userData
-          )
-        this.#syncApi[kHandleDiscoveryKey](discoveryKey, protomux)
-      },
-    })
-    return replicationStream
-  }
-
```
Comment (EvanHahn) [inline src/mapeo-project.js:600]: No longer needed either.

File: src/mapeo-manager.js [hunk 4]
```diff
@@ -208,22 +207,6 @@ export class MapeoManager extends TypedEmitter {
...
-   *
-   * @param {boolean} isInitiator
-   */
-  [kManagerReplicate](isInitiator) {
-    const noiseStream = new NoiseSecretStream(isInitiator, undefined, {
-      keyPair: this.#keyManager.getIdentityKeypair(),
-    })
-    return this.#replicate(noiseStream)
-  }
-
```
Comment (EvanHahn) [inline src/mapeo-manager.js:226]: No longer needed.

**Comment by EvanHahn** [comment]
This is in draft because I'm not sure we want to do this. I worry there's some historical context I'm missing.

**Review by gmaclennan — APPROVED** [review]
looks good, good to simplify things.

**Pull request merged**